pkg/lib/infra/whatsapp/models.go

```diff
@@ -132,6 +132,13 @@ type WhatsappAPIErrorResponse struct {
 	Errors *[]WhatsappAPIErrorDetail `json:"errors,omitempty"`
 }
 
+func (r *WhatsappAPIErrorResponse) FirstErrorCode() (int, bool) {
+	if r.Errors != nil && len(*r.Errors) > 0 {
```

Comment:
The type `*[]WhatsappAPIErrorDetail` is pointer to a slice, and a slice is by definition a pointer to some underlying array. Let's change it to `[]WhatsappAPIErrorDetail` to eliminate one level of indirection.